apps/laboratory/tests/shared/validators/ModalValidator.ts

```diff
@@ -15,13 +15,41 @@ export class ModalValidator {
     this.page = page
   }
 
+  async getLocator(selector: string, timeout = 5000, interval = 500): Promise<Locator> {
+    const start = Date.now()
+    while (Date.now() - start < timeout) {
+      const element = this.page.locator(selector)
+      if (element) {
+        return element
+      }
+      // eslint-disable-next-line no-await-in-loop
+      await this.page.waitForTimeout(interval) // Wait before rechecking
+    }
+    throw new Error(`Element ${selector} did not become visible within ${timeout}ms`)
+  }
+
+  async getByTestId(testId: string | RegExp, timeout = 5000, interval = 500): Promise<Locator> {
+    const start = Date.now()
+    while (Date.now() - start < timeout) {
+      const element = this.page.getByTestId(testId)
+      if (element) {
+        return element
+      }
+      // eslint-disable-next-line no-await-in-loop
+      await this.page.waitForTimeout(interval) // Wait before rechecking
+    }
+    throw new Error(
+      `Element with data-testid="${testId}" did not become visible within ${timeout}ms`
```

Comment:
Should we make a utils like `getElement` or something that has the timing part of these?

Reply:
wdym? to move these into the shared/utils?